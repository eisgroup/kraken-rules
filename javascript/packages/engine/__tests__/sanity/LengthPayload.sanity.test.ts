/*
 *  Copyright 2019 EIS Ltd and/or one of its affiliates.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

import { sanityMocks } from "./_AutoPolicyObject.mocks";
import { sanityEngine } from "./_SanityEngine";

describe("Engine Sanity Length Payload Test", () => {
    const { empty, inValid, valid } = sanityMocks;
    it("should execute 'LengthAutoPolicy' entrypoint with valid data", () => {
        const results = sanityEngine.evaluate(valid(), "LengthAutoPolicy");
        expect(results).k_toMatchResultsStats({ total: 2, critical: 0 });
        expect(results).k_toHaveExpressionsFailures(0);
        expect(results).k_toMatchResultsSnapshots();
    });
    it("should execute 'LengthAutoPolicy' entrypoint with empty data", () => {
        const emptyData = empty();
        emptyData.parties!.push({
            "id": "1",
            "cd": "Party",
            "name": "Party"
        } as {});
        const results = sanityEngine.evaluate(emptyData, "LengthAutoPolicy");
        expect(results).k_toMatchResultsStats({ total: 2, critical: 0 });
        expect(results).k_toHaveExpressionsFailures(0);
        expect(results).k_toMatchResultsSnapshots();
    });
    it("should execute 'LengthAutoPolicy' entrypoint with not valid data", () => {
        const notValidData = inValid();
        notValidData!.parties![0]["relationToPrimaryInsured"] = "Related to Primary Insured";
        notValidData!.parties!.push({
            "id": "1",
            "cd": "Party",
            "name": "Party",
            "relationToPrimaryInsured": "Not Related to Primary Insured"
        } as {});
        const results = sanityEngine.evaluate(notValidData, "LengthAutoPolicy");
        expect(results).k_toMatchResultsStats({ total: 2, critical: 2 });
        expect(results).k_toHaveExpressionsFailures(0);
        expect(results).k_toMatchResultsSnapshots();
    });
    it("should execute 'LengthAutoPolicy' entrypoint with equal length data", () => {
        const validData = valid();
        const results = sanityEngine.evaluate(validData, "LengthAutoPolicy");
        expect(results).k_toMatchResultsStats({ total: 2, critical: 0 });
        expect(results).k_toHaveExpressionsFailures(0);
        expect(results).k_toMatchResultsSnapshots();
    });
    it("should execute 'LengthAutoPolicy' entrypoint and reduce results", () => {
        const results = sanityEngine.evaluate(valid(), "LengthAutoPolicy");
        expect(results).k_toMatchResultsStats({ total: 2, critical: 0 });
        expect(results).k_toHaveExpressionsFailures(0);
        expect(results).k_toMatchResultsSnapshots();
    });
});
