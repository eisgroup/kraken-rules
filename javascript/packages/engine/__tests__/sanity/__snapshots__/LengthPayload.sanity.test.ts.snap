// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Engine Sanity Length Payload Test should execute 'LengthAutoPolicy' entrypoint and reduce results 1`] = `
Object {
  "entryPointResults": Object {
    "evaluationTimestamp": Any<Date>,
    "results": Object {
      "Party:15:relationToPrimaryInsured": Object {
        "contextFieldInfo": DefaultContextFieldInfo {
          "contextId": "15",
          "contextName": "Party",
          "fieldName": "relationToPrimaryInsured",
          "fieldPath": "relationToPrimaryInsured",
          "fieldType": "STRING",
        },
        "ruleResults": Array [
          Object {
            "conditionEvaluationResult": DefaultConditionEvaluationResult {
              "conditionEvaluation": "APPLICABLE",
              "error": undefined,
            },
            "kind": 2,
            "overrideInfo": Object {
              "overridable": false,
              "overrideApplicable": false,
              "overrideGroup": undefined,
            },
            "payloadResult": Object {
              "length": 20,
              "message": Object {
                "errorCode": "code",
                "errorMessage": "String contains more charactes than 20",
                "rawTemplateVariables": Array [],
                "templateVariables": Array [],
              },
              "success": true,
              "type": 16,
              "validationSeverity": "critical",
            },
            "ruleInfo": DefaultRuleInfo {
              "context": "Party",
              "payloadtype": "LENGTH",
              "ruleName": "R0149A",
              "targetPath": "relationToPrimaryInsured",
            },
          },
        ],
      },
      "Party:18:relationToPrimaryInsured": Object {
        "contextFieldInfo": DefaultContextFieldInfo {
          "contextId": "18",
          "contextName": "Party",
          "fieldName": "relationToPrimaryInsured",
          "fieldPath": "relationToPrimaryInsured",
          "fieldType": "STRING",
        },
        "ruleResults": Array [
          Object {
            "conditionEvaluationResult": DefaultConditionEvaluationResult {
              "conditionEvaluation": "APPLICABLE",
              "error": undefined,
            },
            "kind": 2,
            "overrideInfo": Object {
              "overridable": false,
              "overrideApplicable": false,
              "overrideGroup": undefined,
            },
            "payloadResult": Object {
              "length": 20,
              "message": Object {
                "errorCode": "code",
                "errorMessage": "String contains more charactes than 20",
                "rawTemplateVariables": Array [],
                "templateVariables": Array [],
              },
              "success": true,
              "type": 16,
              "validationSeverity": "critical",
            },
            "ruleInfo": DefaultRuleInfo {
              "context": "Party",
              "payloadtype": "LENGTH",
              "ruleName": "R0149A",
              "targetPath": "relationToPrimaryInsured",
            },
          },
        ],
      },
    },
    "ruleTimezoneId": "Europe/Vilnius",
  },
  "fieldResults": Object {
    "Party:15:relationToPrimaryInsured": Object {
      "fieldType": "STRING",
      "id": "Party:15:relationToPrimaryInsured",
      "info": Object {
        "getContextInstanceId": [Function],
        "getContextName": [Function],
      },
      "isDisabled": undefined,
      "isHidden": undefined,
      "resolvedTargetPath": "relationToPrimaryInsured",
      "ruleResults": Array [
        Object {
          "errorCode": "code",
          "errorMessage": "String contains more charactes than 20",
          "isFailed": false,
          "isOverridable": false,
          "isOverridden": false,
          "payloadResult": Object {
            "length": 20,
            "message": Object {
              "errorCode": "code",
              "errorMessage": "String contains more charactes than 20",
              "rawTemplateVariables": Array [],
              "templateVariables": Array [],
            },
            "success": true,
            "type": 16,
            "validationSeverity": "critical",
          },
          "rawTemplateVariables": Array [],
          "ruleName": "R0149A",
          "severity": "critical",
          "templateVariables": Array [],
        },
      ],
    },
    "Party:18:relationToPrimaryInsured": Object {
      "fieldType": "STRING",
      "id": "Party:18:relationToPrimaryInsured",
      "info": Object {
        "getContextInstanceId": [Function],
        "getContextName": [Function],
      },
      "isDisabled": undefined,
      "isHidden": undefined,
      "resolvedTargetPath": "relationToPrimaryInsured",
      "ruleResults": Array [
        Object {
          "errorCode": "code",
          "errorMessage": "String contains more charactes than 20",
          "isFailed": false,
          "isOverridable": false,
          "isOverridden": false,
          "payloadResult": Object {
            "length": 20,
            "message": Object {
              "errorCode": "code",
              "errorMessage": "String contains more charactes than 20",
              "rawTemplateVariables": Array [],
              "templateVariables": Array [],
            },
            "success": true,
            "type": 16,
            "validationSeverity": "critical",
          },
          "rawTemplateVariables": Array [],
          "ruleName": "R0149A",
          "severity": "critical",
          "templateVariables": Array [],
        },
      ],
    },
  },
}
`;

exports[`Engine Sanity Length Payload Test should execute 'LengthAutoPolicy' entrypoint with empty data 1`] = `
Object {
  "entryPointResults": Object {
    "evaluationTimestamp": Any<Date>,
    "results": Object {
      "Party:1:relationToPrimaryInsured": Object {
        "contextFieldInfo": DefaultContextFieldInfo {
          "contextId": "1",
          "contextName": "Party",
          "fieldName": "relationToPrimaryInsured",
          "fieldPath": "relationToPrimaryInsured",
          "fieldType": "STRING",
        },
        "ruleResults": Array [
          Object {
            "conditionEvaluationResult": DefaultConditionEvaluationResult {
              "conditionEvaluation": "APPLICABLE",
              "error": undefined,
            },
            "kind": 2,
            "overrideInfo": Object {
              "overridable": false,
              "overrideApplicable": false,
              "overrideGroup": undefined,
            },
            "payloadResult": Object {
              "length": 20,
              "message": Object {
                "errorCode": "code",
                "errorMessage": "String contains more charactes than 20",
                "rawTemplateVariables": Array [],
                "templateVariables": Array [],
              },
              "success": true,
              "type": 16,
              "validationSeverity": "critical",
            },
            "ruleInfo": DefaultRuleInfo {
              "context": "Party",
              "payloadtype": "LENGTH",
              "ruleName": "R0149A",
              "targetPath": "relationToPrimaryInsured",
            },
          },
        ],
      },
      "Party:3:relationToPrimaryInsured": Object {
        "contextFieldInfo": DefaultContextFieldInfo {
          "contextId": "3",
          "contextName": "Party",
          "fieldName": "relationToPrimaryInsured",
          "fieldPath": "relationToPrimaryInsured",
          "fieldType": "STRING",
        },
        "ruleResults": Array [
          Object {
            "conditionEvaluationResult": DefaultConditionEvaluationResult {
              "conditionEvaluation": "APPLICABLE",
              "error": undefined,
            },
            "kind": 2,
            "overrideInfo": Object {
              "overridable": false,
              "overrideApplicable": false,
              "overrideGroup": undefined,
            },
            "payloadResult": Object {
              "length": 20,
              "message": Object {
                "errorCode": "code",
                "errorMessage": "String contains more charactes than 20",
                "rawTemplateVariables": Array [],
                "templateVariables": Array [],
              },
              "success": true,
              "type": 16,
              "validationSeverity": "critical",
            },
            "ruleInfo": DefaultRuleInfo {
              "context": "Party",
              "payloadtype": "LENGTH",
              "ruleName": "R0149A",
              "targetPath": "relationToPrimaryInsured",
            },
          },
        ],
      },
    },
    "ruleTimezoneId": "Europe/Vilnius",
  },
  "fieldResults": Object {
    "Party:1:relationToPrimaryInsured": Object {
      "fieldType": "STRING",
      "id": "Party:1:relationToPrimaryInsured",
      "info": Object {
        "getContextInstanceId": [Function],
        "getContextName": [Function],
      },
      "isDisabled": undefined,
      "isHidden": undefined,
      "resolvedTargetPath": "relationToPrimaryInsured",
      "ruleResults": Array [
        Object {
          "errorCode": "code",
          "errorMessage": "String contains more charactes than 20",
          "isFailed": false,
          "isOverridable": false,
          "isOverridden": false,
          "payloadResult": Object {
            "length": 20,
            "message": Object {
              "errorCode": "code",
              "errorMessage": "String contains more charactes than 20",
              "rawTemplateVariables": Array [],
              "templateVariables": Array [],
            },
            "success": true,
            "type": 16,
            "validationSeverity": "critical",
          },
          "rawTemplateVariables": Array [],
          "ruleName": "R0149A",
          "severity": "critical",
          "templateVariables": Array [],
        },
      ],
    },
    "Party:3:relationToPrimaryInsured": Object {
      "fieldType": "STRING",
      "id": "Party:3:relationToPrimaryInsured",
      "info": Object {
        "getContextInstanceId": [Function],
        "getContextName": [Function],
      },
      "isDisabled": undefined,
      "isHidden": undefined,
      "resolvedTargetPath": "relationToPrimaryInsured",
      "ruleResults": Array [
        Object {
          "errorCode": "code",
          "errorMessage": "String contains more charactes than 20",
          "isFailed": false,
          "isOverridable": false,
          "isOverridden": false,
          "payloadResult": Object {
            "length": 20,
            "message": Object {
              "errorCode": "code",
              "errorMessage": "String contains more charactes than 20",
              "rawTemplateVariables": Array [],
              "templateVariables": Array [],
            },
            "success": true,
            "type": 16,
            "validationSeverity": "critical",
          },
          "rawTemplateVariables": Array [],
          "ruleName": "R0149A",
          "severity": "critical",
          "templateVariables": Array [],
        },
      ],
    },
  },
}
`;

exports[`Engine Sanity Length Payload Test should execute 'LengthAutoPolicy' entrypoint with equal length data 1`] = `
Object {
  "entryPointResults": Object {
    "evaluationTimestamp": Any<Date>,
    "results": Object {
      "Party:15:relationToPrimaryInsured": Object {
        "contextFieldInfo": DefaultContextFieldInfo {
          "contextId": "15",
          "contextName": "Party",
          "fieldName": "relationToPrimaryInsured",
          "fieldPath": "relationToPrimaryInsured",
          "fieldType": "STRING",
        },
        "ruleResults": Array [
          Object {
            "conditionEvaluationResult": DefaultConditionEvaluationResult {
              "conditionEvaluation": "APPLICABLE",
              "error": undefined,
            },
            "kind": 2,
            "overrideInfo": Object {
              "overridable": false,
              "overrideApplicable": false,
              "overrideGroup": undefined,
            },
            "payloadResult": Object {
              "length": 20,
              "message": Object {
                "errorCode": "code",
                "errorMessage": "String contains more charactes than 20",
                "rawTemplateVariables": Array [],
                "templateVariables": Array [],
              },
              "success": true,
              "type": 16,
              "validationSeverity": "critical",
            },
            "ruleInfo": DefaultRuleInfo {
              "context": "Party",
              "payloadtype": "LENGTH",
              "ruleName": "R0149A",
              "targetPath": "relationToPrimaryInsured",
            },
          },
        ],
      },
      "Party:18:relationToPrimaryInsured": Object {
        "contextFieldInfo": DefaultContextFieldInfo {
          "contextId": "18",
          "contextName": "Party",
          "fieldName": "relationToPrimaryInsured",
          "fieldPath": "relationToPrimaryInsured",
          "fieldType": "STRING",
        },
        "ruleResults": Array [
          Object {
            "conditionEvaluationResult": DefaultConditionEvaluationResult {
              "conditionEvaluation": "APPLICABLE",
              "error": undefined,
            },
            "kind": 2,
            "overrideInfo": Object {
              "overridable": false,
              "overrideApplicable": false,
              "overrideGroup": undefined,
            },
            "payloadResult": Object {
              "length": 20,
              "message": Object {
                "errorCode": "code",
                "errorMessage": "String contains more charactes than 20",
                "rawTemplateVariables": Array [],
                "templateVariables": Array [],
              },
              "success": true,
              "type": 16,
              "validationSeverity": "critical",
            },
            "ruleInfo": DefaultRuleInfo {
              "context": "Party",
              "payloadtype": "LENGTH",
              "ruleName": "R0149A",
              "targetPath": "relationToPrimaryInsured",
            },
          },
        ],
      },
    },
    "ruleTimezoneId": "Europe/Vilnius",
  },
  "fieldResults": Object {
    "Party:15:relationToPrimaryInsured": Object {
      "fieldType": "STRING",
      "id": "Party:15:relationToPrimaryInsured",
      "info": Object {
        "getContextInstanceId": [Function],
        "getContextName": [Function],
      },
      "isDisabled": undefined,
      "isHidden": undefined,
      "resolvedTargetPath": "relationToPrimaryInsured",
      "ruleResults": Array [
        Object {
          "errorCode": "code",
          "errorMessage": "String contains more charactes than 20",
          "isFailed": false,
          "isOverridable": false,
          "isOverridden": false,
          "payloadResult": Object {
            "length": 20,
            "message": Object {
              "errorCode": "code",
              "errorMessage": "String contains more charactes than 20",
              "rawTemplateVariables": Array [],
              "templateVariables": Array [],
            },
            "success": true,
            "type": 16,
            "validationSeverity": "critical",
          },
          "rawTemplateVariables": Array [],
          "ruleName": "R0149A",
          "severity": "critical",
          "templateVariables": Array [],
        },
      ],
    },
    "Party:18:relationToPrimaryInsured": Object {
      "fieldType": "STRING",
      "id": "Party:18:relationToPrimaryInsured",
      "info": Object {
        "getContextInstanceId": [Function],
        "getContextName": [Function],
      },
      "isDisabled": undefined,
      "isHidden": undefined,
      "resolvedTargetPath": "relationToPrimaryInsured",
      "ruleResults": Array [
        Object {
          "errorCode": "code",
          "errorMessage": "String contains more charactes than 20",
          "isFailed": false,
          "isOverridable": false,
          "isOverridden": false,
          "payloadResult": Object {
            "length": 20,
            "message": Object {
              "errorCode": "code",
              "errorMessage": "String contains more charactes than 20",
              "rawTemplateVariables": Array [],
              "templateVariables": Array [],
            },
            "success": true,
            "type": 16,
            "validationSeverity": "critical",
          },
          "rawTemplateVariables": Array [],
          "ruleName": "R0149A",
          "severity": "critical",
          "templateVariables": Array [],
        },
      ],
    },
  },
}
`;

exports[`Engine Sanity Length Payload Test should execute 'LengthAutoPolicy' entrypoint with not valid data 1`] = `
Object {
  "entryPointResults": Object {
    "evaluationTimestamp": Any<Date>,
    "results": Object {
      "Party:1:relationToPrimaryInsured": Object {
        "contextFieldInfo": DefaultContextFieldInfo {
          "contextId": "1",
          "contextName": "Party",
          "fieldName": "relationToPrimaryInsured",
          "fieldPath": "relationToPrimaryInsured",
          "fieldType": "STRING",
        },
        "ruleResults": Array [
          Object {
            "conditionEvaluationResult": DefaultConditionEvaluationResult {
              "conditionEvaluation": "APPLICABLE",
              "error": undefined,
            },
            "kind": 2,
            "overrideInfo": Object {
              "overridable": false,
              "overrideApplicable": false,
              "overrideGroup": undefined,
            },
            "payloadResult": Object {
              "length": 20,
              "message": Object {
                "errorCode": "code",
                "errorMessage": "String contains more charactes than 20",
                "rawTemplateVariables": Array [],
                "templateVariables": Array [],
              },
              "success": false,
              "type": 16,
              "validationSeverity": "critical",
            },
            "ruleInfo": DefaultRuleInfo {
              "context": "Party",
              "payloadtype": "LENGTH",
              "ruleName": "R0149A",
              "targetPath": "relationToPrimaryInsured",
            },
          },
        ],
      },
      "Party:3:relationToPrimaryInsured": Object {
        "contextFieldInfo": DefaultContextFieldInfo {
          "contextId": "3",
          "contextName": "Party",
          "fieldName": "relationToPrimaryInsured",
          "fieldPath": "relationToPrimaryInsured",
          "fieldType": "STRING",
        },
        "ruleResults": Array [
          Object {
            "conditionEvaluationResult": DefaultConditionEvaluationResult {
              "conditionEvaluation": "APPLICABLE",
              "error": undefined,
            },
            "kind": 2,
            "overrideInfo": Object {
              "overridable": false,
              "overrideApplicable": false,
              "overrideGroup": undefined,
            },
            "payloadResult": Object {
              "length": 20,
              "message": Object {
                "errorCode": "code",
                "errorMessage": "String contains more charactes than 20",
                "rawTemplateVariables": Array [],
                "templateVariables": Array [],
              },
              "success": false,
              "type": 16,
              "validationSeverity": "critical",
            },
            "ruleInfo": DefaultRuleInfo {
              "context": "Party",
              "payloadtype": "LENGTH",
              "ruleName": "R0149A",
              "targetPath": "relationToPrimaryInsured",
            },
          },
        ],
      },
    },
    "ruleTimezoneId": "Europe/Vilnius",
  },
  "fieldResults": Object {
    "Party:1:relationToPrimaryInsured": Object {
      "fieldType": "STRING",
      "id": "Party:1:relationToPrimaryInsured",
      "info": Object {
        "getContextInstanceId": [Function],
        "getContextName": [Function],
      },
      "isDisabled": undefined,
      "isHidden": undefined,
      "resolvedTargetPath": "relationToPrimaryInsured",
      "ruleResults": Array [
        Object {
          "errorCode": "code",
          "errorMessage": "String contains more charactes than 20",
          "isFailed": true,
          "isOverridable": false,
          "isOverridden": false,
          "payloadResult": Object {
            "length": 20,
            "message": Object {
              "errorCode": "code",
              "errorMessage": "String contains more charactes than 20",
              "rawTemplateVariables": Array [],
              "templateVariables": Array [],
            },
            "success": false,
            "type": 16,
            "validationSeverity": "critical",
          },
          "rawTemplateVariables": Array [],
          "ruleName": "R0149A",
          "severity": "critical",
          "templateVariables": Array [],
        },
      ],
    },
    "Party:3:relationToPrimaryInsured": Object {
      "fieldType": "STRING",
      "id": "Party:3:relationToPrimaryInsured",
      "info": Object {
        "getContextInstanceId": [Function],
        "getContextName": [Function],
      },
      "isDisabled": undefined,
      "isHidden": undefined,
      "resolvedTargetPath": "relationToPrimaryInsured",
      "ruleResults": Array [
        Object {
          "errorCode": "code",
          "errorMessage": "String contains more charactes than 20",
          "isFailed": true,
          "isOverridable": false,
          "isOverridden": false,
          "payloadResult": Object {
            "length": 20,
            "message": Object {
              "errorCode": "code",
              "errorMessage": "String contains more charactes than 20",
              "rawTemplateVariables": Array [],
              "templateVariables": Array [],
            },
            "success": false,
            "type": 16,
            "validationSeverity": "critical",
          },
          "rawTemplateVariables": Array [],
          "ruleName": "R0149A",
          "severity": "critical",
          "templateVariables": Array [],
        },
      ],
    },
  },
}
`;

exports[`Engine Sanity Length Payload Test should execute 'LengthAutoPolicy' entrypoint with valid data 1`] = `
Object {
  "entryPointResults": Object {
    "evaluationTimestamp": Any<Date>,
    "results": Object {
      "Party:15:relationToPrimaryInsured": Object {
        "contextFieldInfo": DefaultContextFieldInfo {
          "contextId": "15",
          "contextName": "Party",
          "fieldName": "relationToPrimaryInsured",
          "fieldPath": "relationToPrimaryInsured",
          "fieldType": "STRING",
        },
        "ruleResults": Array [
          Object {
            "conditionEvaluationResult": DefaultConditionEvaluationResult {
              "conditionEvaluation": "APPLICABLE",
              "error": undefined,
            },
            "kind": 2,
            "overrideInfo": Object {
              "overridable": false,
              "overrideApplicable": false,
              "overrideGroup": undefined,
            },
            "payloadResult": Object {
              "length": 20,
              "message": Object {
                "errorCode": "code",
                "errorMessage": "String contains more charactes than 20",
                "rawTemplateVariables": Array [],
                "templateVariables": Array [],
              },
              "success": true,
              "type": 16,
              "validationSeverity": "critical",
            },
            "ruleInfo": DefaultRuleInfo {
              "context": "Party",
              "payloadtype": "LENGTH",
              "ruleName": "R0149A",
              "targetPath": "relationToPrimaryInsured",
            },
          },
        ],
      },
      "Party:18:relationToPrimaryInsured": Object {
        "contextFieldInfo": DefaultContextFieldInfo {
          "contextId": "18",
          "contextName": "Party",
          "fieldName": "relationToPrimaryInsured",
          "fieldPath": "relationToPrimaryInsured",
          "fieldType": "STRING",
        },
        "ruleResults": Array [
          Object {
            "conditionEvaluationResult": DefaultConditionEvaluationResult {
              "conditionEvaluation": "APPLICABLE",
              "error": undefined,
            },
            "kind": 2,
            "overrideInfo": Object {
              "overridable": false,
              "overrideApplicable": false,
              "overrideGroup": undefined,
            },
            "payloadResult": Object {
              "length": 20,
              "message": Object {
                "errorCode": "code",
                "errorMessage": "String contains more charactes than 20",
                "rawTemplateVariables": Array [],
                "templateVariables": Array [],
              },
              "success": true,
              "type": 16,
              "validationSeverity": "critical",
            },
            "ruleInfo": DefaultRuleInfo {
              "context": "Party",
              "payloadtype": "LENGTH",
              "ruleName": "R0149A",
              "targetPath": "relationToPrimaryInsured",
            },
          },
        ],
      },
    },
    "ruleTimezoneId": "Europe/Vilnius",
  },
  "fieldResults": Object {
    "Party:15:relationToPrimaryInsured": Object {
      "fieldType": "STRING",
      "id": "Party:15:relationToPrimaryInsured",
      "info": Object {
        "getContextInstanceId": [Function],
        "getContextName": [Function],
      },
      "isDisabled": undefined,
      "isHidden": undefined,
      "resolvedTargetPath": "relationToPrimaryInsured",
      "ruleResults": Array [
        Object {
          "errorCode": "code",
          "errorMessage": "String contains more charactes than 20",
          "isFailed": false,
          "isOverridable": false,
          "isOverridden": false,
          "payloadResult": Object {
            "length": 20,
            "message": Object {
              "errorCode": "code",
              "errorMessage": "String contains more charactes than 20",
              "rawTemplateVariables": Array [],
              "templateVariables": Array [],
            },
            "success": true,
            "type": 16,
            "validationSeverity": "critical",
          },
          "rawTemplateVariables": Array [],
          "ruleName": "R0149A",
          "severity": "critical",
          "templateVariables": Array [],
        },
      ],
    },
    "Party:18:relationToPrimaryInsured": Object {
      "fieldType": "STRING",
      "id": "Party:18:relationToPrimaryInsured",
      "info": Object {
        "getContextInstanceId": [Function],
        "getContextName": [Function],
      },
      "isDisabled": undefined,
      "isHidden": undefined,
      "resolvedTargetPath": "relationToPrimaryInsured",
      "ruleResults": Array [
        Object {
          "errorCode": "code",
          "errorMessage": "String contains more charactes than 20",
          "isFailed": false,
          "isOverridable": false,
          "isOverridden": false,
          "payloadResult": Object {
            "length": 20,
            "message": Object {
              "errorCode": "code",
              "errorMessage": "String contains more charactes than 20",
              "rawTemplateVariables": Array [],
              "templateVariables": Array [],
            },
            "success": true,
            "type": 16,
            "validationSeverity": "critical",
          },
          "rawTemplateVariables": Array [],
          "ruleName": "R0149A",
          "severity": "critical",
          "templateVariables": Array [],
        },
      ],
    },
  },
}
`;
